app/views/step_by_step_pages/show/_actions.erb

```diff
@@ -1,6 +1,11 @@
 <% preview_link = render "govuk_publishing_components/components/button", {
   text: "Preview",
-  href: preview_url(@step_by_step_page.slug, auth_bypass_id: @step_by_step_page.auth_bypass_id),
+  href: preview_url(
+    @step_by_step_page.slug,
+    auth_bypass_id: @step_by_step_page.auth_bypass_id,
+    step_by_step_id: @step_by_step_page.content_id,
+    user_id: current_user.uid,
+  ),
```

Comment:
I'd suggest that we have a particular helper function for `step_by_step` preview urls to make this more concise, since we're passing in a few arguments from a step by step page - do we have a view helper for step_by_step pages?

Could be something like:

```
def step_by_step_page_preview_url(step_by_step_page, user = nil)
  user ||= current_user
  token_payload = {
    "sub" => step_by_step_page.auth_bypass_id,
    "iss" => user.uid,
    "iat" => Time.zone.now.to_i,
    "exp" => 1.month.from_now.to_i,
    "draft_asset_manager_access" => true,
    "content_id" => step_by_step_page.content_id,
  }
  token = JWT.encode(token_payload, ENV["JWT_SECRET"], "HS256")
  
  "#{Plek.new.external_url_for('draft-origin')}/#{step_by_step_page.slug}?token=#{token}"
end
```

I'm not sure we need the whole cascade of the existing methods - do other things use them? I say this from a slight prejudice that the JwtHelper class is not a real class so smells a bit too me 🙂

Reply:
Agreed 👍 thanks for confirming I'm on the right track!